src/test/java/org/snomed/snowstorm/core/data/services/ConceptServiceTest.java

```diff
@@ -2383,6 +2384,193 @@ void testCreateNewExtensionComponent_EffectiveTimeIsNull_DefaultModuleId() throw
+
+		// Create type of Vehicle (purposely without Axiom)
+		concept = new Concept()
+				.addDescription(new Description("Car (vehicle)").setTypeId(FSN).setCaseSignificance(ci).setAcceptabilityMap(intPreferred))
+				.addDescription(new Description("Car").setTypeId(SYNONYM).setCaseSignificance(ci).setAcceptabilityMap(intPreferred))
+				.addRelationship(new Relationship(ISA, vehicleId));
+		concept = conceptService.create(concept, intMain);
+		String carId = concept.getConceptId();
+
+		// Manually create Axiom incorrectly (formatted back to front)
+		ReferenceSetMember axiom = new ReferenceSetMember(CORE_MODULE, OWL_AXIOM_REFERENCE_SET, concept.getConceptId());
+		String backToFront = String.format("SubClassOf(:%s :%s)", vehicleId, carId); // Note: vehicleId -> carId
+		axiom.setAdditionalField("owlExpression", backToFront);
+		axiom = referenceSetMemberService.createMember(intMain, axiom);
+		String invalidAxiomId = axiom.getId();
+
+		// Assert formatted incorrectly
+		axiom = referenceSetMemberService.findMember(intMain, axiom.getMemberId());
+		String owlExpression = axiom.getAdditionalField("owlExpression");
+		assertEquals(backToFront, owlExpression);
+
+		// Version error
+		codeSystem = codeSystemService.find("SNOMEDCT");
+		codeSystemService.createVersion(codeSystem, 20240201, "20240201");
+
+		// Fix mistake (replace invalid with valid)
+		axiom = referenceSetMemberService.findMember(intMain, invalidAxiomId);
+		axiom.setActive(false);
+		axiom.updateEffectiveTime();
+		referenceSetMemberService.updateMember(intMain, axiom);
+
+		axiom = new ReferenceSetMember(CORE_MODULE, OWL_AXIOM_REFERENCE_SET, concept.getConceptId());
+		String frontToBack = String.format("SubClassOf(:%s :%s)", carId, vehicleId);  // Note: carId -> vehicleId
+		axiom.setAdditionalField("owlExpression", frontToBack);
+		axiom = referenceSetMemberService.createMember(intMain, axiom);
+		String validAxiomId = axiom.getId();
+
+		// Version correction
+		codeSystem = codeSystemService.find("SNOMEDCT");
+		codeSystemService.createVersion(codeSystem, 20240301, "20240301");
+
+		// Add synonym
+		concept = conceptService.find(carId, intMain);
+		concept.addDescription(new Description("Motorised vehicle").setTypeId(SYNONYM).setCaseSignificance(ci).setAcceptabilityMap(intPreferred));
+		conceptService.update(concept, intMain);
+
+		// Assert inactive axiom hasn't changed
+		Axiom invalidAxiom = concept.getClassAxioms().stream().filter(cA -> cA.getAxiomId().equals(invalidAxiomId)).collect(Collectors.toList()).get(0);
+		assertEquals(20240301, invalidAxiom.getEffectiveTimeI());
+		assertEquals(backToFront, invalidAxiom.getReferenceSetMember().getAdditionalField("owlExpression"));
+	}
+
+	@Test
+	void testActivatingInvalidAxiomThrowsException() throws ServiceException {
+		String intMain = "MAIN";
+		Map<String, String> intPreferred = Map.of(US_EN_LANG_REFSET, descriptionAcceptabilityNames.get(PREFERRED), GB_EN_LANG_REFSET, descriptionAcceptabilityNames.get(PREFERRED));
```

Comment:
@pgwilliams this tests that if an invalid inactive axiom is reactivated, an exception is thrown. The exception bubbles up from a transitive closure problem.